src/main/java/org/assertj/core/api/AbstractIterableAssert.java

```diff
@@ -2754,6 +2754,51 @@ public SELF allSatisfy(Consumer<? super ELEMENT> requirements) {
     return myself;
   }
 
+  /**
+   * Verifies that the all the given consumers can be satisfied by elements in the iterable with
+   * an element at most satisfies one consumer. No order requirement.
+   * <p>
+   * This assertion is to test whether there are enough elements in the iterable that can satisfy all the consumers.
+   * <p>
+   * Grouping assertions example :
+   * <pre><code class='java'> // second constructor parameter is the light saber color
+   * List&lt;String&gt; actual = newArrayList("Luke", "Leia", "Yoda");
+   *
+   * // this assertion succeeds:
+   * assertThat(actual).satisfy((s --&gt; {
+   *       assertThat(s).doesNotContain("L");
+   *     }, s --&gt; {
+   *       assertThat(s.length()).isEqualTo(4);
+   *       assertThat(s).doesNotContain("a");
+   *     }));
+   *
+   * // these assertions fail:
+   * assertThat(actual).satisfy((s --&gt; {
+   *       assertThat(s).doesNotContain("L");
+   *     }, s --&gt; {
+   *       assertThat(s).doesNotContain("L");
+   *     }));
+   *
+   * assertThat(actual).satisfy((s --&gt; {
+   *       assertThat(s).contains("L");
+   *     }, s --&gt; {
+   *       assertThat(s).contains("L");
+   *     }, s --&gt; {
+   *       assertThat(s).contains("L");
+   *     }));</code></pre>
```

Comment:
Would it be more readable to name the list "starWarsCharacterNames" or "names" instead of "actual"?
Similarly, could the lambda arguments renamed from "s" to "name"?

In the code examples, the ```consumers``` argument of the ```satisfy``` method has two parentheses. Can the inner parenthesis be removed (i.e., ```satisfy(s --&gt; ``` instead of ```satisfy((s --&gt; ```)?